persistence-modules/spring-data-jpa-query-4/src/main/java/com/baeldung/spring/data/jpa/maxvalue/Employee.java

```diff
@@ -0,0 +1,61 @@
+package com.baeldung.spring.data.jpa.maxvalue;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import java.util.Objects;
+
+@Entity
+public class Employee {
+    @Id
+    @GeneratedValue
+    private Integer id;
+    private String name;
+    private Long salary;
+
+    public Employee() {
+    }
+
+    public Employee(Integer id, String name, Long salary) {
+        this.id = id;
+        this.name = name;
+        this.salary = salary;
+    }
+
+    public Integer getId() {
+        return id;
+    }
+
+    public void setId(Integer id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Long getSalary() {
+        return salary;
+    }
+
+    public void setSalary(Long salary) {
+        this.salary = salary;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Employee employee = (Employee) o;
+        return Objects.equals(id, employee.id) && Objects.equals(name, employee.name) && Objects.equals(salary, employee.salary);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, name, salary);
+    }
```

Comment:
Are these necessary for the production code to work or just used for the test assertions?

Reply:
not necessary, just auto-generated via IJ, removed